### ISSUE TITLE
[BUG] negative indexing selection fails in columns containing nulls

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#1494]
**Describe the bug**
Negative index selection fails in columns containing nulls (not just through iloc). It fails whether or not the index value is "before" or "after" the null.

```
import cudf
import pandas as pd

pdf = pd.DataFrame({'a':[0,1,2,3,4],
                     'b':[0,1,2,3,4],
                     'c':[0,1,2,3,4],
                     'd':[0,1,2, None, 4],
                    })
df = cudf.from_pandas(pdf)

print(df.c.iloc[-3])
print(df.d.iloc[-3])
2
None
```

**Comment by awthomp** [comment]
This is impeding my work with [Whitewater](https://github.com/awthomp/whitewater) where I need to handle arbitrary data sizes.

The current method of data handling is to expect a csv file of data with the first N-1 columns consisting of our feature X matrix and the last column consisting of our 'y' observation vector, e.g.

```
import numpy as np
import cudf

M = 1e3

df = cudf.DataFrame()
df['col1'] = np.random.rand(int(M))
df['col2'] = np.random.rand(int(M))
df['y'] = np.random.rand(int(M))
```

Where:
```
X = df.iloc[: , :-1]
y = df.iloc[:, -1]
```

iloc is currently unable to handle:
1. grabbing columns by index (it returns the whole dataframe)
2. grabbing groups of columns with splicing, e.g. df.iloc[:, 0:2]

My workaround in my project is (in this example):
```
X, y = df.loc[:, ['col1', 'col2']], df.loc[:, 'y']
```

But this obviously requires a-priori knowledge of the column names.

**Comment by beckernick** [comment]
Resolved by #1622.
